iOSScratchApp/MBPlayerManager.swift

```diff
@@ -82,6 +84,13 @@ class MBPlayerManager: NSObject {
         }
     }
     
+    func routeHasChanged(notification: NSNotification) {
+        if let userInfo = notification.userInfo, let reason = AVAudioSessionRouteChangeReason(rawValue: userInfo[AVAudioSessionRouteChangeReasonKey] as! UInt) where reason == .OldDeviceUnavailable {
+            self.playState = .Paused
+            self.delegate?.isPausing()
+        }
+    }
```

Comment:
No the "OldDeviceUnavailable" is for when the headphones are unplugged that output device is unavailable and then allows me to put the app in the paused state because by default iOS pauses any app playback when that happens. The only problem was that the UI was not notified of this system level pause...until now